app/models/auth/voms.rb

```diff
@@ -41,11 +41,21 @@ def parse_hash_groups!(hash)
         raise Errors::AuthenticationError, 'voms group env variable is not set' unless hash.key?('GRST_VOMS_FQANS')
         groups = Hash.new { |h, k| h[k] = [] }
         hash['GRST_VOMS_FQANS'].split(';').each do |line|
-          matches = line.match(VOMS_GROUP_REGEXP)
-          groups[matches[:group]] << matches[:role] if matches && matches[:role] != 'NULL'
+          group = parse_group(line)
+          groups.merge!(group) { |_, oldval, newval| oldval + newval } if group
         end
         groups.map { |key, value| { id: key, roles: value } }
       end
+
+      def parse_group(line)
+        matches = line.match(VOMS_GROUP_REGEXP)
+        return unless matches
```

Comment:
I would definitely raise an error instead of just returning. If something fishy is going on, I would much rather know about it as soon as possible.